projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -2780,3 +2792,84 @@ function wpcom_launchpad_get_latest_draft_id() {
 
 	return $cached_draft_id;
 }
+
+/**
+ * Checks if the current site primary domain is a WPCOM domain.
+ *
+ * @return bool Will return true if the primary domain is a WPCOM domain.
+ */
+function wpcom_launchpad_is_primary_domain_wpcom() {
+	if ( ! ( new Automattic\Jetpack\Status\Host() )->is_wpcom_platform() ) {
+		return false;
+	}
+
+	if ( ! class_exists( 'Domain_Mapping' ) || ! class_exists( 'WPCOM_Domain' ) ) {
+		return false;
+	}
+
+	$blog_id = get_current_blog_id();
+
+	// @phan-suppress-next-line PhanUndeclaredClassMethod -- Being checked before being called.
+	$primary_domain_mapping = Domain_Mapping::find_primary_by_blog_id( $blog_id );
+	$is_wpcom_domain        = true;
+
+	if ( null !== $primary_domain_mapping ) {
+		// @phan-suppress-next-line PhanUndeclaredClassMethod -- Being checked before being called.
+		$wpcom_domain = new WPCOM_Domain( $primary_domain_mapping->get_domain_name() );
+		// @phan-suppress-next-line PhanUndeclaredClassMethod
+		$is_wpcom_domain = $wpcom_domain->is_wpcom_tld();
+	}
+
+	return $is_wpcom_domain;
+}
+
+/**
+ * Check if the `check_ssl_status` task is complete.
+ *
+ * @return bool
+ */
+function wpcom_launchpad_is_ssl_task_completed() {
+	// Only run on WPCOM platform.
+	if ( ! ( new Automattic\Jetpack\Status\Host() )->is_wpcom_platform() ) {
+		return false;
+	}
+
+	$task_id = 'check_ssl_status';
+
+	// If the task is already complete, return true.
+	if ( wpcom_launchpad_is_task_option_completed( array( 'id' => $task_id ) ) ) {
+		return true;
+	}
+
+	if ( ! class_exists( 'Domain_Mapping' ) || ! class_exists( 'Domain_Certificate_Flag_Group' ) || ! class_exists( 'WPCOM\Container\DI' ) || ! class_exists( 'Domain_Certificate_Flags_Manager' ) ) {
+		return false;
+	}
+
+	if ( ! wpcom_launchpad_is_primary_domain_wpcom() ) {
```

Comment:
Doing it this way means we are calling `Domain_Mapping::find_primary_by_blog_id( $blog_id );` three times now with every request. That's three more DB queries.

Can we cache that result somewhere during the lifecycle of that request?

would adding something like that help?

```
function wpcom_launchpad_get_primary_domain( int $blog_id ) {
    static $last_blog_id;
    static $primary_domain;
    
    if ( isset($last_blog_id) && isset( $primary_domain ) && $last_blog_id === $blog_id ) {
        return $primary_domain;
    }


	if ( ! class_exists( 'Domain_Mapping' ) || ! class_exists( 'WPCOM_Domain' ) ) {
		return null;
	}

	$primary_domain_mapping = Domain_Mapping::find_primary_by_blog_id( $blog_id );

    if ( null === $primary_domain_mapping ) {
        return null;
    }

    $primary_domain = new WPCOM_Domain( $primary_domain_mapping->get_domain_name() );
    
    return $primary_domain;
}
```

This is just pseudo-code that I wrote here, please test and adapt as you see fit :)